### PR TITLE
Add test for dynamic indexing of matrices and vectors

### PR DESCRIPTION
**Pull request by Oletus**
This test goes through cases where bugs might slip into certain type of
code transformations done to replace dynamic indexing with static
indexing as is required on a HLSL backend of WebGL.

### CODE REVIEW

File: sdk/tests/conformance2/glsl3/vector-dynamic-indexing.html [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+<script id="fshaderIndexMatrixTwice" type="x-shader/x-fragment">#version 300 es
+precision mediump float;
+
+out vec4 my_FragColor;
+
+uniform int u_zero;
+
+void main() {
+    mat2 m = mat2(0.0, 0.0, 0.0, 1.0);
+    float f = m[u_zero + 1][u_zero + 1];
```
Comment (zhenyao) [inline sdk/tests/conformance2/glsl3/vector-dynamic-indexing.html]: This got me thinking, what is the behavior if dynamic index is out of range?  For static index, it's an error at compile time.

**Comment by zhenyao** [comment]
The test itself LGTM

**Comment by kenrussell** [comment]
Thanks for the thorough test Olli and thanks Mo for reviewing it. I didn't have time to go through it in detail but at a high level it LGTM too.

File: sdk/tests/conformance2/glsl3/vector-dynamic-indexing.html (hunk 1 above)
Comment (Oletus) [inline sdk/tests/conformance2/glsl3/vector-dynamic-indexing.html]: Same answer as on ANGLE code review: The base spec says that this is undefined behavior. KHR_robust_buffer_access_behavior says that either zero or any value from the active program's memory may be returned. We can consider specifying this more tightly than what KHR_robust_buffer_access_behavior does, but possible costs need to be considered carefully. ANGLE already clamps indices, but that might not be strictly necessary on all backends if the goal is only memory security.

Comment (zhenyao) [inline sdk/tests/conformance2/glsl3/vector-dynamic-indexing.html]: Thanks for clarifying this.

**Pull request merged**